auth/src/main/java/org/aerogear/auth/IUserPrincipal.java

```diff
@@ -0,0 +1,30 @@
+package org.aerogear.auth;
+
+import org.aerogear.auth.credentials.TokenCredentials;
+
+import java.security.Principal;
+
+/**
+ * Public interface for user principals.
+ */
+public interface IUserPrincipal extends Principal {
+
+    /**
+     * Checks if the user has the specified role.
+     * @param role role to be checked
+     * @return true or false
+     */
+    boolean hasRole(IRole role);
+
+    /**
+     * Returns the username
+     * @return the username
+     */
+    String getName();
```

Comment:
`getUsername()` might be a bit more accurate

Reply:
the interface currently extends the java.security.Principal class.
Since a Principal can be different things (not only a User) it forces the implementation of the getName method.
If you believe we could add the getUsername method or remove the inheritance from the Principal interface.

Reply:
Ah, I see. I am just thinking of a use case where a username could be used as sort of an identifier or primary key for some functionality, and the Full Name would be used in the UI on the user's profile.